src/main/java/com/iota/iri/network/pipeline/TransactionProcessingPipelineImpl.java

```diff
@@ -135,6 +143,12 @@ private void addStage(String name, BlockingQueue<ProcessingContext> queue,
                             receivedStageQueue.put(payload.getRight());
                             break;
                         case BROADCAST:
+                            try {
+                                BroadcastPayload broadcastPayload = (BroadcastPayload) ctx.getPayload();
+                                txSolidifier.addToSolidificationQueue(broadcastPayload.getTransactionViewModel().getHash());
```

Comment:
It is kind of weird to me that if a transaction is quickly solid it may be still be added to the solidification queue...

Reply:
Again this was to bolster broadcasting speed, which in turn increases the rate that transaction requests are sent to neighbours. By placing transactions that were quickly solid into the queue as well, it'll push it into the broadcast queue to send out another transaction request. I agree it comes off as hacky, but the way I see it, it's hacky to be reliant on the broadcast queue for transaction requesting. 

I think a new stage for handling specifically transaction requests would be beneficial and would circumvent the need for forcing out as many broadcasts as possible, and would allow for only solid transactions to be placed into the broadcast queue as defined in #1512 . This can be it's own issue and PR and i'll be mentioning it in an issue i'll be creating. I'll link that issue to this one as well for reference.